R/appMetadata.R

```diff
@@ -6,32 +6,27 @@ appMetadata <- function(appDir,
                         isCloudServer = FALSE,
                         metadata = list()) {
 
-  if (is.null(appFiles)) {
-    appFiles <- bundleFiles(appDir)
-  }
+  appFiles <- standardizeAppFiles(appDir, appFiles)
 
-  if (!is.null(contentCategory)) {
-    assetTypeName <- contentCategory
-  } else if (!is.null(appPrimaryDoc)) {
-    assetTypeName <- "document"
+  # User has supplied quarto path or quarto package has supplied metadata
+  # https://github.com/quarto-dev/quarto-r/blob/08caf0f42504e7/R/publish.R#L117-L121
+  hasQuarto <- !is.null(quarto) || hasQuartoMetadata(metadata)
+
+  # Generally we want to infer appPrimaryDoc from appMode, but there's one
+  # special case
+  if (!is.null(appPrimaryDoc) &&
+      tolower(tools::file_ext(appPrimaryDoc)) == "r") {
+    appMode <- "shiny"
   } else {
-    assetTypeName <- "application"
+    # Inference only uses top-level files
+    rootFiles <- appFiles[dirname(appFiles) == "."]
```

Comment:
We should explain that `appFiles` should contain relative paths referencing files beneath `appDir`. Should `file.path(appDir, someAppFile)` not exist, it is silently dropped (see `rsconnect:::explodeFiles`).

I forgot this detail and needed to run a few `rsconnect:::standardizeAppFiles` examples to prove that it would only return relative paths.